testsuite/features/step_definitions/common_steps.rb

```diff
@@ -1161,3 +1161,35 @@ def server_secret
   xpath = "//td[contains(text(), '#{action_name}')]/ancestor::tr[contains(@class, 'list-row-even')]/td/div/button/#{button}"
   raise "xpath: #{xpath} not found" unless find(:xpath, xpath).click
 end
+
+
+# todo is this the right place?
+# common constants for SSH bootstrap with SSH key auth test
+key_filename = 'id_rsa_bootstrap-passphrase_linux.pub'
+## authorized_keys paths on the client
+auth_keys_path = '/root/.ssh/authorized_keys'
+auth_keys_sav_path = '/root/.ssh/authorized_keys.sav'
+
+When(/^I backup the SSH authorized_keys file of host "([^"]*)"$/) do |host|
+  target = get_target(host)
+  _, ret_code = target.run("cp #{auth_keys_path} #{auth_keys_sav_path}")
+  raise 'error backing up authorized_keys on host' if ret_code.nonzero?
+end
+
+And(/^I add pre\-generated SSH public key to authorized_keys of host "([^"]*)"$/) do |host|
+  target = get_target(host)
+  ret_code = file_inject(
+    target,
+    File.dirname(__FILE__) + '/../upload_files/ssh_keypair/' + key_filename,
```

Comment:
or two string interpolations with `#{...}`(not sure it's a good idea, feel free to ignore)

Reply:
`common_steps` contain every possible way to combine path now, so I'll leave this for now since it works ;) (best way would probably be `File.join(File.dirname(__FILE__), 'something')`, but I don't want to break already working code).

Reply:
yes, it might be we have been creative here... :blush: